frontend/lib/Providers/CompareEndpointsModel.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:adpv_frontend/Repository/AbstractEndpointRepository.dart';
+import 'package:adpv_frontend/Repository/RestClient.dart';
+import 'package:dio/dio.dart';
+import 'package:flutter/cupertino.dart';
+
+import '../Models/Endpoint.dart';
+import '../Models/EndpointSummary.dart';
+
+class EndpointModel extends ChangeNotifier {
+  Map<String, EndpointSummary> endpointSummaryMap = {};
+  Map<String, Endpoint> endpointsMap = {};
+  Map<String, bool> selectedChips = {};
+  List<String> commonFields = [];
+  List<String> selectedEndpoints = [];
+  AbstractEndpointRepository repository = RestClient(Dio());
+
+  EndpointModel();
+
+  void selectChips(String label, bool value) {
+    selectedChips[label] = value;
+    if (value) {
+      selectedChips[label] = true;
+    } else {
+      selectedChips[label] = false;
+    }
+    notifyListeners();
+  }
+
+  void updateCommonFields() {
+    if (selectedEndpoints.isNotEmpty){
+      Map<String, int> counter = {};
+      List fields = endpointsMap[selectedEndpoints[0]]!.data.dataList[0].keys.toList();
+      for(String s in fields){
+        for(Endpoint e in endpointsMap.values){
+          if(e.hasField(s)) {
+            counter[s] = counter[s] == null ? 1 : counter[s]! + 1;
```

Comment:
Chyba będzie krócej i to samo, ale mogę się mylić
```dart
counter[s] = (counter[s] ?? 0) + 1;
```

Reply:
prawda, kwestia gustu, zostawiam moje. Operator ?? jest spoko, ale jakoś go nie lubię

Reply:
Okej, jak wolisz